### ISSUE TITLE
windows terminal breaks on `ctrl+C`

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#1279]
### Describe the bug

https://discord.com/channels/1212416027611365476/1317009958113644554

### Reproduction

I don't have Windows, see discord.

### Steps to reproduce

```sh
pnpm dev
```

### System Info

```shell
TODO
```

### Used Package Manager

pnpm

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/wxt-dev/wxt/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://wxt.dev/guide/installation.html).
- [X] Check that there isn't [already an issue](https://github.com/wxt-dev/wxt/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/wxt-dev/wxt/discussions) or join our [Discord Chat Server](https://discord.gg/ZFsZqGery9).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by nishu-murmu** [comment]
[Here's the loom video](https://www.loom.com/share/7fc46cf75a2e4aca9e3f422b1282e1b3?sid=066fe681-4a1f-48cf-87b7-be44950fb0d6) from my window's 10 system.

I don't think so it seems to be the case with readline (still skeptical tho).
Cause in the video you can see after terminating server. My powershell seems to be loading properly.
But that is also very inconsistent.
But after running `pnpm test` command it crashes.

**Comment by aklinker1** [comment]
@nishu-murmu So with my readline change, you can press `ctrl+C` twice and everything works just fine? Still need to figure out why the prompt get's messed up, but that's progress, right?

**Comment by nishu-murmu** [comment]
@aklinker1 
Yeah, in my powershell it seems to be working.
In windows it does works but very inconsistent, sometimes I also face that crash issue where powershell gets useless and am not able to enter any command.
In linux it is showing something like this.
![open-source](https://github.com/user-attachments/assets/3d88a448-1db4-43f6-bb2d-35941856f4f5)

Windows is more problematic.

Maybe we need to handle that exit code 130 thing? I saw some similar code in `vitest`'s codebase. They were programmatically handling that case.
What do you think?

**Comment by aklinker1** [comment]
> They were programmatically handling that case.

Can you share where they are doing that?

**Comment by nishu-murmu** [comment]
Here,
https://github.com/vitest-dev/vitest/blob/main/packages/vitest/src/node/stdin.ts#L49-L68

But they seem to be using raw input mode.
Can you also look around a bit and see if it's possible to use raw input mode.

**Comment by aklinker1** [comment]
Yeah, we specifically chose not to use raw input mode because then you have to manually handle all keyboard shortcuts for interrupting the process.

**Comment by nishu-murmu** [comment]
Yeah I know, I'm not suggesting raw input mode. I'm pitching maybe we can do something for windows specifically.
They also OS specific code.

**Comment by aklinker1** [comment]
Oh I see, sorry yes, that makes sense. Though it means we have to manually handle an edge case, which I wanted to avoid :/

@nishu-murmu Did you ever explore older versions of WXT and find the version that introduced this behavior? I don't think I ever heard confirmation from anyone that v0.19.19 actually introduced this.

**Comment by nishu-murmu** [comment]
This is powershell terminal

https://www.loom.com/share/f914b2f34f994551b630b231708be473?sid=f9191019-8916-4064-b9e9-72fbd2df767b

Here is the tested video for version 0.19.18
It seems to be working fine.
After pressing `Ctrl + C` the powershell terminal is working fine.
But yeah pressing twice works.

**Comment by aklinker1** [comment]
Does the "-> wxt-demo" usually show up when asking to confirm stopping the process? I haven't used powershell in a while but previously, that prompt would show on its own line.

**Comment by nishu-murmu** [comment]
Yes,
It shows up when asking for confirmation.
This works fine initially.
For any version.
But doing this more than twice or thrice breaks the windows terminal
https://www.loom.com/share/f914b2f34f994551b630b231708be473?sid=34c3b762-0b10-447f-9d9e-033fbbb074e9

**Comment by aklinker1** [comment]
Doesn't look like its work to me. When you type "Y", shouldn't it stop the process and give you a new blank line to type on? I understand that pressing Ctrl+C again makes it work, but you shouldn't have to press it again after typing "Y"

**Comment by aklinker1** [comment]
@ianho7 No. I should probably disable this feature on windows. I need someone with windows to contribute a working solution.

**Comment by GunsyX** [comment]
Ran into this issue, using cmd instead of powershell avoids this crash, I configured default terminal for my extension repository to be CMD via adding 
```
"settings": {
	"terminal.integrated.defaultProfile.windows": "Command Prompt"
}
```
to the .code-workspace file.

**Comment by PatrykKuniczak** [comment]
Terminating on windows with `CTRL+C` is nightmare.
If i terminated process then i hadn't able to rerun script, only i can do is close terminal and reopen it and then i'm able but not everytime sometime i have info `buildc is running, waiting for finish` or sth like that.

Also i have problem with terminate `pnpm test` it's probably worst than `dev`, because if i terminated then i hadn't able to rerun and i need to wait a while to all previous run ends.

https://github.com/user-attachments/assets/c844ced8-b819-44dc-a9f8-0a6f1891ec74

PS: I think it's issue with `buildc` package
